ice-adapter/src/main/java/com/faforever/iceadapter/ice/GameSession.java

```diff
@@ -126,29 +129,53 @@ public CompletableFuture<OptionalDouble> load(String host) throws Exception {
                 }
 
                 urls.stream()
-                        .map(IceServer.urlPattern::matcher)
-                        .filter(Matcher::matches)
-                        .forEach(matcher -> {
-                            String host = matcher.group("host");
-                            int port = Optional.ofNullable(matcher.group("port")).map(Integer::parseInt).orElse(3478);
-                            Transport transport = Optional.ofNullable(matcher.group("transport")).map(Transport::parse).orElse(Transport.UDP);
-
-                            TransportAddress address = new TransportAddress(host, port, transport);
-                            (matcher.group("protocol").equals("stun") ? iceServer.getStunAddresses() : iceServer.getTurnAddresses()).add(address);
-
-                            if (IceAdapter.PING_COUNT > 0) {
-                                iceServer.setRoundTripTime(hostRTTCache.getUnchecked(host));
-                            }
-
-                            coturnServers.add(new CoturnServer("n/a", host, port, null));
-                        });
+                    .map(stringUrl -> {
+                        try {
+                            return new URL(stringUrl);
+                        } catch (Exception e) {
+                            log.warn("Invalid ICE server URL: {}", stringUrl);
+                            return null;
+                        }
+                    })
+                    .filter(Objects::nonNull)
+                    .filter(url -> VALID_PROTOCOLS.contains(url.getProtocol()))
```

Comment:
Shouldn't we also log a warn for invalid protocols?

Reply:
I guess we can. I just didn't see too much utility to that

Reply:
Given the amount of weird errors we were hunting so far, we should log everything that seems odd

Reply:
Done